catalog/catalog-domain/src/main/java/com/cgdpd/library/catalog/domain/book/dto/SearchBookCriteria.java

```diff
@@ -24,4 +26,20 @@ public SearchBookCriteria(Optional<String> bookTitle,
         this.publicationYearLessThan = actualOrEmpty(publicationYearLessThan);
         this.publicationYearGreaterThan = actualOrEmpty(publicationYearGreaterThan);
     }
+
+    public static SearchBookCriteria empty() {
+        return new SearchBookCriteria(Optional.empty(), Optional.empty(),
+              Optional.empty(), Optional.empty(), Optional.empty());
+    }
+
+    public boolean matches(Book book) {
+        return bookTitle.map(title -> title.equals(book.title())).orElse(true)
+              && authorName.map(name -> name.equals(book.authorId().value().toString()))
```

Comment:
This method does should not be here, besides why are we comparing authorName with authorId?